### PR TITLE
add nat gw e2e

### PR DESCRIPTION
**Pull request by bobz965**
- [ ] Make sure you have followed [Kube-OVN Code Style](https://github.com/kubeovn/kube-ovn/blob/master/CODE_STYLE.md).

add iptables nat gw eip fip snat dnat e2e
add ovn nat gw eip fip snat dnat e2e
differentiate non ovn subnet with ovn subnet clearly
non ovn subnet should have ready conditon
simplicy subnet enable lb

### What type of this PR
- Tests

Fixes #(https://github.com/kubeovn/kube-ovn/issues/2589)

### <samp>🤖 Generated by Copilot at dcdc421</samp>

This pull request improves the controller logic, status updates, and e2e testing for various kube-ovn resources, such as subnets, eips, vlans, and iptables. It simplifies the code, fixes some bugs, and adds new features and clients for different providers and CRDs. It also updates the types, deep copy methods, and default values for some fields.

### <samp>🤖 Generated by Copilot at dcdc421</samp>

> _We patch the status, we simplify the code_
> _We test the features, we fix the bugs_
> _We handle the subnets, we support the CRDs_
> _We are the masters of the e2e framework_

### <samp>🤖 Generated by Copilot at dcdc421</samp>

*  Remove pointer type for EnableLb field in SubnetSpec type and simplify the logic of handling the default value and nil pointer dereference errors ([link](https://github.com/kubeovn/kube-ovn/pull/2639/files?diff=unified&w=0#diff-69f75d53361877cc02ab6f413f306486d29287c6400b92b71c4c74f018f3234aL140-R142), [link](https://github.com/kubeovn/kube-ovn/pull/2639/files?diff=unified&w=0#diff-06f8e5bc84406eb8a89c72c345fc4ff5f144e480103ad55d681f7567f0f85515L1532-L1536), [link](https://github.com/kubeovn/kube-ovn/pull/2639/files?diff=unified&w=0#diff-92aa62435bc2b1b2be34f478f650597068bcf35b8d5975d119f91cf78ac54988L134-R134), [link](https://github.com/kubeovn/kube-ovn/pull/2639/files?diff=unified&w=0#diff-3742843fd11c637b26eae183fb8a0c4eeff96bbfb18a8db1e0ded0c33733f205L97-R97), [link](https://github.com/kubeovn/kube-ovn/pull/2639/files?diff=unified&w=0#diff-3742843fd11c637b26eae183fb8a0c4eeff96bbfb18a8db1e0ded0c33733f205L332-R348), [link](https://github.com/kubeovn/kube-ovn/pull/2639/files?diff=unified&w=0#diff-3742843fd11c637b26eae183fb8a0c4eeff96bbfb18a8db1e0ded0c33733f205L705-R736))
* Add vpc and provider fields to SubnetSpec type and skip the reconciliation and validation logic for non-ovn subnets ([link](https://github.com/kubeovn/kube-ovn/pull/2639/files?diff=unified&w=0#diff-3742843fd11c637b26eae183fb8a0c4eeff96bbfb18a8db1e0ded0c33733f205L316-R321), [link](https://github.com/kubeovn/kube-ovn/pull/2639/files?diff=unified&w=0#diff-3742843fd11c637b26eae183fb8a0c4eeff96bbfb18a8db1e0ded0c33733f205L487-R499), [link](https://github.com/kubeovn/kube-ovn/pull/2639/files?diff=unified&w=0#diff-3742843fd11c637b26eae183fb8a0c4eeff96bbfb18a8db1e0ded0c33733f205R557-R587), [link](https://github.com/kubeovn/kube-ovn/pull/2639/files?diff=unified&w=0#diff-3742843fd11c637b26eae183fb8a0c4eeff96bbfb18a8db1e0ded0c33733f205R593), [link](https://github.com/kubeovn/kube-ovn/pull/2639/files?diff=unified&w=0#diff-3742843fd11c637b26eae183fb8a0c4eeff96bbfb18a8db1e0ded0c33733f205L579-L592), [link](https://github.com/kubeovn/kube-ovn/pull/2639/files?diff=unified&w=0#diff-3742843fd11c637b26eae183fb8a0c4eeff96bbfb18a8db1e0ded0c33733f205L602-L609), [link](https://github.com/kubeovn/kube-ovn/pull/2639/files?diff=unified&w=0#diff-f3c9b1195f52feef2061addba23cdc2815ad06445a5f63316661197212307ddfL258-R259), [link](https://github.com/kubeovn/kube-ovn/pull/2639/files?diff=unified&w=0#diff-f3c9b1195f52feef2061addba23cdc2815ad06445a5f63316661197212307ddfL264-R270), [link](https://github.com/kubeovn/kube-ovn/pull/2639/files?diff=unified&w=0#diff-f3c9b1195f52feef2061addba23cdc2815ad06445a5f63316661197212307ddfL273-R282))
* Remove unnecessary finalizers for ovn eip CRs and add Type field to ovn eip status ([link](https://github.com/kubeovn/kube-ovn/pull/2639/files?diff=unified&w=0#diff-28764fafc63230ed4d8301d469d3a770071b3db0759ff3c58121653fcf0334ceL266-L269), [link](https://github.com/kubeovn/kube-ovn/pull/2639/files?diff=unified&w=0#diff-28764fafc63230ed4d8301d469d3a770071b3db0759ff3c58121653fcf0334ceL376-L379), [link](https://github.com/kubeovn/kube-ovn/pull/2639/files?diff=unified&w=0#diff-28764fafc63230ed4d8301d469d3a770071b3db0759ff3c58121653fcf0334ceR425), [link](https://github.com/kubeovn/kube-ovn/pull/2639/files?diff=unified&w=0#diff-ae8a8b5d44d277a4bb96635ee325fe54b48e8bc10b24a79d4bb8e211b57131ecL836-R838), [link](https://github.com/kubeovn/kube-ovn/pull/2639/files?diff=unified&w=0#diff-ae8a8b5d44d277a4bb96635ee325fe54b48e8bc10b24a79d4bb8e211b57131ecL911-L921))
* Use Patch method instead of UpdateStatus method for updating the status of provider networks to avoid conflicts with the controller-runtime manager ([link](https://github.com/kubeovn/kube-ovn/pull/2639/files?diff=unified&w=0#diff-8e705abac23a0d3a04e999fbbb84ae8907471e95ffc24a9317487f41ee2e11f4R20-R28), [link](https://github.com/kubeovn/kube-ovn/pull/2639/files?diff=unified&w=0#diff-7405fc764106a56c3d8c53ca59401ba6b0c285e3a136518b6be936880cbcd8d0L300-R307))
* Add e2e framework clients for iptables and ovn CRDs and implement the methods for creating, updating, deleting, and waiting for the CRs ([link](https://github.com/kubeovn/kube-ovn/pull/2639/files?diff=unified&w=0#diff-af3a26e61a4ea8976c87a2b53f1588f3847d2e16a13103e7d9eae8cd667e14bdR1-R166), [link](https://github.com/kubeovn/kube-ovn/pull/2639/files?diff=unified&w=0#diff-a93889515fa94ec885fca2cbe615fd2e2f23913893ad72208ad9b02b7d8a552bR1-R166), [link](https://github.com/kubeovn/kube-ovn/pull/2639/files?diff=unified&w=0#diff-98890dda301ed54549d84343026658b76d76d9dd5f2e2e800cd9a2d2543d13feR1-R163), [link](https://github.com/kubeovn/kube-ovn/pull/2639/files?diff=unified&w=0#diff-72bff85ba17e518669e16f041d64784fc06dee69cc945fdf32afa1be874b7630R1-R163), [link](https://github.com/kubeovn/kube-ovn/pull/2639/files?diff=unified&w=0#diff-1fd340c6ac03534744a9a4031b8eee8d0fad027c4d57b647eefc8247eb45cf82R1-R169),  [link](https://github.com/kubeovn/kube-ovn/pull/2639/files?diff=unified&w=0#diff-c3fefe1ef80254801f8ffeabe13da702e419ac012ff2cd6080632f4893639ff1R1-R166), [link](https://github.com/kubeovn/kube-ovn/pull/2639/files?diff=unified&w=0#diff-b8690b937d8f43d4d3745a072dd98d7e8d8fa81bf76832aae78d36147fe08823R1-R166))
* Add e2e tests for iptables-vpc-nat-gw and ovn-vpc-nat-gw features and add the corresponding targets in the Makefile ([link](https://github.com/kubeovn/kube-ovn/pull/2639/files?diff=unified&w=0#diff-c6c608a44221cc490b39781deca27f76d95887d10f75895d2a302a9cbc8a0f6aL137-R154), [link](https://github.com/kubeovn/kube-ovn/pull/2639/files?diff=unified&w=0#diff-c354587e36151f73f4a4da62c22d612510bee633ada8a52ed16b89c7185f0283R13-R14), [link](https://github.com/kubeovn/kube-ovn/pull/2639/files?diff=unified&w=0#diff-c354587e36151f73f4a4da62c22d612510bee633ada8a52ed16b89c7185f0283L29-R31), [link](https://github.com/kubeovn/kube-ovn/pull/2639/files?diff=unified&w=0#diff-c354587e36151f73f4a4da62c22d612510bee633ada8a52ed16b89c7185f0283R127-R137))
* Improve the logging and debugging of the e2e tests by adding log messages when the CRs reach the desired condition status ([link](https://github.com/kubeovn/kube-ovn/pull/2639/files?diff=unified&w=0#diff-7c453ce6d7a2eac0da282cb8104da75464fdf606ed6c64b9b65a576c9e934fcbL108-R116), [link](https://github.com/kubeovn/kube-ovn/pull/2639/files?diff=unified&w=0#diff-f3c9b1195f52feef2061addba23cdc2815ad06445a5f63316661197212307ddfR175))
* Simplify the logic of waiting for the ovn eip to disappear by using the Get method instead of the List method ([link](https://github.com/kubeovn/kube-ovn/pull/2639/files?diff=unified&w=0#diff-7c453ce6d7a2eac0da282cb8104da75464fdf606ed6c64b9b65a576c9e934fcbL154-R151))
* Reorder the assignment of the Spec fields and check the v4ip parameter instead of the mac parameter for updating the iptables eip CR ([link](https://github.com/kubeovn/kube-ovn/pull/2639/files?diff=unified&w=0#diff-c2fca6de8b38cf14047d762d1029e969197a5fd0d5172062dce753f1e9613c92L548-R550))
* Remove duplicate status patching for the IP field of the iptables eip status ([link](https://github.com/kubeovn/kube-ovn/pull/2639/files?diff=unified&w=0#diff-c2fca6de8b38cf14047d762d1029e969197a5fd0d5172062dce753f1e9613c92L236-L239))
* Remove unused patchEipIP function ([link](https://github.com/kubeovn/kube-ovn/pull/2639/files?diff=unified&w=0#diff-c2fca6de8b38cf14047d762d1029e969197a5fd0d5172062dce753f1e9613c92L675-L699))
* Use lowercase for consistency with other log messages in the ovn-eip.go file ([link](https://github.com/kubeovn/kube-ovn/pull/2639/files?diff=unified&w=0#diff-7c453ce6d7a2eac0da282cb8104da75464fdf606ed6c64b9b65a576c9e934fcbL126-R130), [link](https://github.com/kubeovn/kube-ovn/pull/2639/files?diff=unified&w=0#diff-7c453ce6d7a2eac0da282cb8104da75464fdf606ed6c64b9b65a576c9e934fcbL135-R140))
* Add import statements for the net, strings, and controllerutil packages in the iptables-eip.go file ([link](https://github.com/kubeovn/kube-ovn/pull/2639/files?diff=unified&w=0#diff-c2fca6de8b38cf14047d762d1029e969197a5fd0d5172062dce753f1e9613c92R6-R8), [link](https://github.com/kubeovn/kube-ovn/pull/2639/files?diff=unified&w=0#diff-c2fca6de8b38cf14047d762d1029e969197a5fd0d5172062dce753f1e9613c92L13-R16))
* Remove import statement for the net package in the iptables-eip.go file ([link](https://github.com/kubeovn/kube-ovn/pull/2639/files?diff=unified&w=0#diff-c2fca6de8b38cf14047d762d1029e969197a5fd0d5172062dce753f1e9613c92L13-R16))

### CODE REVIEW

File: pkg/controller/subnet.go [hunk 1]
```diff
@@ -542,11 +544,43 @@ func (c *Controller) handleAddOrUpdateSubnet(key string) error {
...
+
+	if err = util.ValidateSubnet(*subnet); err != nil {
+		klog.Errorf("failed to validate subnet %s, %v", subnet.Name, err)
+		c.patchSubnetStatus(subnet, "ValidateLogicalSwitchFailed", err.Error())
+		return err
+	} else {
+		c.patchSubnetStatus(subnet, "ValidateLogicalSwitchSuccess", "")
+	}
+
+	if subnet.Spec.Vpc == "" && !isOvnSubnet(subnet) {
```
Comment (hongzhen-ma) [inline pkg/controller/subnet.go:569]: 需要专门处理非 ovn 子网的情况吗？下边有检查，非ovn 子网返回了

File: pkg/controller/subnet.go [hunk 2]
```diff
@@ -576,20 +610,6 @@ func (c *Controller) handleAddOrUpdateSubnet(key string) error {
...
-		return err
-	}
-
-	if err := c.ipam.AddOrUpdateSubnet(subnet.Name, subnet.Spec.CIDRBlock, subnet.Spec.Gateway, subnet.Spec.ExcludeIps); err != nil {
-		return err
-	}
-
 	if err := c.reconcileU2OInterconnectionIP(subnet); err != nil {
 		klog.Errorf("failed to reconcile underlay subnet %s to overlay interconnection %v", subnet.Name, err)
 		return err
```
Comment (hongzhen-ma) [inline pkg/controller/subnet.go:615]: 非 ovn 子网，在下一行检查返回了。如果不需要 reconcile 非 subnet的子网，把这个函数，放在 ovn子网检查后边，就可以了吧？

File: pkg/controller/subnet.go [hunk 3]
```diff
@@ -329,14 +330,12 @@ func formatSubnet(subnet *kubeovnv1.Subnet, c *Controller) error {
 		}
 	}
 
-	if subnet.Spec.EnableLb == nil && subnet.Name != c.config.NodeSwitch {
-		changed = true
-		subnet.Spec.EnableLb = &c.config.EnableLb
-	}
```
Comment (bobz965) [inline pkg/controller/subnet.go:335]: 这里的逻辑是，enableLB为空的，不是join子网直接跟随config的配置，这个思路还是全局控制的感觉，且为空直觉上对应false，且管理上创建子网用户指定enable lb即可，不再使用该全局配置了

File: pkg/controller/subnet.go [hunk 4]
```diff
@@ -329,14 +330,12 @@ func formatSubnet(subnet *kubeovnv1.Subnet, c *Controller) error {
...
-	// set join subnet Spec.EnableLb to nil
-	if subnet.Spec.EnableLb != nil && subnet.Name == c.config.NodeSwitch {
-		changed = true
-		subnet.Spec.EnableLb = nil
+	if subnet.Spec.EnableLb {
+		// set non ovn subnet, join subnet Spec.EnableLb to false
+		if !isOvnSubnet(subnet) || subnet.Name == c.config.NodeSwitch {
+			changed = true
+			subnet.Spec.EnableLb = false
+		}
```
[the remark this answers is not in the document]
Comment (bobz965) [reply pkg/controller/subnet.go:338]: 是的，这里防止用户配错了。 这种场景只能为false，配成true的情况下也只能为false。

File: pkg/controller/subnet.go (hunk 1 above)
Comment (bobz965) [reply pkg/controller/subnet.go:569]: 非 ovn 子网 没有condition，之前的代码 subnet 和vpc 处理流程代码交错，我单独把subnet处理流程都拉到vpc前面去了。 我在想我要不要把下面的!isOvnSubnet(subnet) {retrun} 去掉。

Comment (bobz965) [reply pkg/controller/subnet.go:569]: e2e 框架代码 要求所有subnet 都必须有ready 的condition

File: pkg/controller/subnet.go (hunk 2 above)
Comment (bobz965) [reply pkg/controller/subnet.go:615]: 我没删掉下面的检查就是因为我不太清楚u2o 面向的是哪种underlay 子网还是所有underlay子网

Comment (bobz965) [reply pkg/controller/subnet.go:615]: 是的，我把ovn子网检查放到前面去

**Review by ShaPoHun — APPROVED** [review]

File: Makefile.e2e [hunk 5]
```diff
@@ -136,14 +137,23 @@ kube-ovn-lb-svc-conformance-e2e:
 	ginkgo $(GINKGO_PARALLEL_OPT) --randomize-all --always-emit-ginkgo-writer \
 		--focus=CNI:Kube-OVN ./test/e2e/lb-svc/lb-svc.test
 
-.PHONY: kube-ovn-eip-conformance-e2e
-kube-ovn-eip-conformance-e2e:
-	ginkgo build $(E2E_BUILD_FLAGS) ./test/e2e/ovn-eip
+.PHONY: iptables-vpc-nat-gw-conformance-e2e
+iptables-vpc-nat-gw-conformance-e2e:
+	ginkgo build ./test/e2e/iptables-vpc-nat-gw
```
Comment (zhangzujian) [inline Makefile.e2e:142]:
```suggestion
	ginkgo build $(E2E_BUILD_FLAGS) ./test/e2e/iptables-vpc-nat-gw
```

File: dist/images/install.sh [hunk 6]
```diff
@@ -2010,7 +2010,6 @@ spec:
                   not:
                     enum:
                       - int
-                      - external
```
Comment (zhangzujian) [inline dist/images/install.sh:2013]: This change should also be applied to `kubeovn-helm/templates/kube-ovn-crd.yaml`.

File: pkg/controller/vlan.go [hunk 7]
```diff
@@ -297,8 +298,13 @@ func (c *Controller) updateProviderNetworkStatusForVlanDeletion(pn *kubeovnv1.Pr
...
 	newPn := pn.DeepCopy()
 	newPn.Status.Vlans = util.RemoveString(newPn.Status.Vlans, vlan)
-	_, err := c.config.KubeOvnClient.KubeovnV1().ProviderNetworks().UpdateStatus(context.Background(), newPn, metav1.UpdateOptions{})
+	bytes, err := newPn.Status.Bytes()
 	if err != nil {
+		klog.Errorf("failed to marshal provider network status '%s', %v", newPn.Name, err)
+		return err
+	}
+	if _, err := c.config.KubeOvnClient.KubeovnV1().ProviderNetworks().Patch(context.Background(), newPn.Name, types.MergePatchType,
+		bytes, metav1.PatchOptions{}, "status"); err != nil {
```
Comment (zhangzujian) [inline pkg/controller/vlan.go:307]: Why change `UpdateStatus()` to `Patch()`?

File: pkg/apis/kubeovn/v1/types.go [hunk 8]
```diff
@@ -137,9 +137,9 @@ type SubnetSpec struct {
 
 	Acls []Acl `json:"acls,omitempty"`
 
-	U2OInterconnection bool  `json:"u2oInterconnection,omitempty"`
-	EnableLb           *bool `json:"enableLb,omitempty"`
-	EnableEcmp         bool  `json:"enableEcmp,omitempty"`
+	U2OInterconnection bool `json:"u2oInterconnection,omitempty"`
+	EnableLb           bool `json:"enableLb,omitempty"`
```
Comment (zhangzujian) [inline pkg/apis/kubeovn/v1/types.go:141]: We should use `*bool` for compatibility in upgrade scenarios.
@hongzhen-ma What do you think?

Comment (hongzhen-ma) [reply pkg/apis/kubeovn/v1/types.go:141]: 是的。这里的修改是目前还没有达成一致的地方。

File: pkg/controller/vlan.go (hunk 7 above)
Comment (bobz965) [reply pkg/controller/vlan.go:307]: update过程会出现一些更新冲突的情况，patch 应该不会

Comment (bobz965) [reply pkg/controller/vlan.go:307]: 更新 CRD 状态时，使用 Patch 更新更好。

原因是使用 Update 更新会覆盖整个 CRD 对象，包括除状态外的其他字段。而使用 Patch 更新只会更新需要修改的字段，不会影响其他字段，避免了出现未预期的改动。

此外，Patch 更新也有更好的性能表现，因为它只需要传输和处理变化的部分，而不是整个对象。这对于大规模集群和复杂 CRD 对象尤其重要。

因此，建议在更新 CRD 状态时使用 Patch 更新。

File: dist/images/install.sh (hunk 6 above)
Comment (bobz965) [reply dist/images/install.sh:2013]: ok

File: Makefile.e2e [hunk 9]
```diff
@@ -136,14 +137,23 @@ kube-ovn-lb-svc-conformance-e2e:
...
+		--focus=CNI:Kube-OVN ./test/e2e/iptables-vpc-nat-gw/iptables-vpc-nat-gw.test
+
+.PHONY: ovn-vpc-nat-gw-conformance-e2e
+ovn-vpc-nat-gw-conformance-e2e:
+	ginkgo build $(E2E_BUILD_FLAGS) ./test/e2e/ovn-vpc-nat-gw
+	E2E_BRANCH=$(E2E_BRANCH) \
+	E2E_IP_FAMILY=$(E2E_IP_FAMILY) \
+	E2E_NETWORK_MODE=$(E2E_NETWORK_MODE) \
+	ginkgo $(GINKGO_PARALLEL_OPT) --randomize-all --always-emit-ginkgo-writer \
+		--focus=CNI:Kube-OVN ./test/e2e/ovn-vpc-nat-gw/ovn-vpc-nat-gw.test
```
Comment (zhangzujian) [inline Makefile.e2e:156]: These test suites should be added and run in GitHub actions, which is defined in `.github/workflows/build-x86-image.yaml` and `.github/workflows/scheduled-e2e.yaml`.

File: pkg/apis/kubeovn/v1/types.go (hunk 8 above)
Comment (bobz965) [reply pkg/apis/kubeovn/v1/types.go:141]: 后续单独提PR 再定

File: Makefile.e2e (hunk 9 above)
Comment (bobz965) [reply Makefile.e2e:156]: ok

File: pkg/controller/vlan.go (hunk 7 above)
Comment (bobz965) [reply pkg/controller/vlan.go]: 后续再讨论 先回滚

File: test/e2e/iptables-vpc-nat-gw/e2e_test.go [hunk 10]
```diff
@@ -194,27 +181,11 @@ var _ = framework.Describe("[group:iptables-vpc-nat-gw]", func() {
...
 		networkClient := attachNetClient.K8sCniCncfIoV1().NetworkAttachmentDefinitions("kube-system")
-		nad := makeNetworkAttachmentDefinition(networkAttachDefName, nadConfig)
-		_, err = networkClient.Create(context.Background(), nad, metav1.CreateOptions{})
-		framework.ExpectNoError(err, "failed to create")
-
-		nad, err = networkClient.Get(context.Background(), networkAttachDefName, metav1.GetOptions{})
-		ginkgo.By("Got network attachment fefinition " + nad.Name)
+		nad, err := networkClient.Get(context.Background(), networkAttachDefName, metav1.GetOptions{})
 		framework.ExpectNoError(err, "failed to get")
+		ginkgo.By("Got network attachment fefinition " + nad.Name)
```
Comment (zhangzujian) [inline test/e2e/iptables-vpc-nat-gw/e2e_test.go:188]:
```suggestion
		ginkgo.By("Got network attachment definition " + nad.Name)
```

**Review by zhangzujian — APPROVED** [review]

File: yamls/vpc-nat-gw-attachment.yaml [hunk 11]
```diff
@@ -0,0 +1,7 @@
+apiVersion: k8s.cni.cncf.io/v1
+kind: NetworkAttachmentDefinition
+metadata:
+  name: ovn-vpc-external-network
+  namespace: kube-system
+spec:
+  config: '{"cniVersion": "0.3.0","type": "macvlan","master": "eth1","mode": "bridge"}'
```
Comment (hongzhen-ma) [inline yamls/vpc-nat-gw-attachment.yaml:7]: 节点上一般是 eth0 网卡，确认存在 eth1 就可以

File: pkg/controller/subnet.go [hunk 12]
```diff
@@ -542,11 +548,50 @@ func (c *Controller) handleAddOrUpdateSubnet(key string) error {
...
+	if !isOvnSubnet(subnet) {
+		// subnet provider is not ovn, and vpc is empty, should not reconcile
+		c.patchSubnetStatus(subnet, "SetNonOvnSubnetSuccess", "")
+
+		subnet.Status.EnsureStandardConditions()
+		klog.Infof("non ovn subnet %s is ready", subnet.Name)
+		return nil
+	}
+
+	if subnet.Spec.Vpc == "" {
```
Comment (hongzhen-ma) [inline pkg/controller/subnet.go:582]: 非 kube-ovn 管理的subnet，上一步返回了。
kube-ovn 管理的subnet，在format中，给subnet.spec.vpc 默认赋值为 ovn-cluster了
所以还需要这个检查吗

**Review by hongzhen-ma — APPROVED** [review]

File: yamls/vpc-nat-gw-attachment.yaml (hunk 11 above)
Comment (bobz965) [reply yamls/vpc-nat-gw-attachment.yaml:7]: 这个e2e中有检查是否存在eth1，如果不存在会中断，这个如果eth1后续e2e不稳定的话，我后续再调整下。

**Pull request merged**